### PR TITLE
NODE-1679 tests for order v3 united

### PR DESCRIPTION
**Pull request by finnoio**
All written before tests for order v3 are aggregated in one package called "orderV3" + tests for percent and fixed matcher modes.

### CODE REVIEW

File: dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+              .placeOrder(sell)
+              .message
+              .id
+
+            orderStatus(alice, someWavesPair, aliceOrderId, "Filled")
+            orderStatus(bob, someWavesPair, bobOrderId, "Filled")
+
+            node.reservedBalance(alice) shouldBe Map.empty
+            node.reservedBalance(bob) shouldBe Map.empty
+//            node.tradableBalance(alice, someWavesPair) shouldBe aliceTradableBalanceBefore
```
Comment (frekensnork) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala]: delete, uncomment or make todo remark

File: dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala [hunk 2]
```diff
@@ -0,0 +1,346 @@
+package com.wavesplatform.it.sync.orderV3
```
Comment (frekensnork) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala]: I suggest to check also fee if order was partially filled, cancelled
Also what will happen, if we have order in Waves-Usd pair, fee is less than one percent, and total amount of price asset is less than 100? What if order was partiall filled twice, fee of both was less than one percent?

File: dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+          "spending",
+          "receiving"
+        )) {
+          docker.restartNode(node,
+            configWithOrderFeePercent(
+              assetType = percentAssetType,
+              minFeePercent = minFeePercent
+            ))
+          for (amountAsset <- Seq(
+            aliceAsset,
```
Comment (frekensnork) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala]: both assets have 8 decimals, need to check fee measurments in pairs like Waves-Wct with much count of extra zeros in price percent mode

File: dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/FixedModeMatcherTestSuite.scala [hunk 4]
```diff
@@ -0,0 +1,253 @@
+package com.wavesplatform.it.sync.orderV3
```
Comment (frekensnork) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/FixedModeMatcherTestSuite.scala]: Partially filled orders, cancelled orders, rounding of small fees, asset pairs with different decimals cound
See below in "dex-it/src/test/scala/com/wavesplatform/it/sync/orderV3/PercentModeMatcherTestSuite.scala"

**Comment by frekensnork** [comment]
Also there is a conflict

File: dex-it/src/test/scala/com/wavesplatform/it/sync/CancelOrderTestSuite.scala [hunk 5]
```diff
@@ -143,130 +144,130 @@ class CancelOrderTestSuite extends MatcherSuiteBase {
...
-  private def inSeries[A, B](xs: Seq[A])(f: A => Future[B]): Future[Seq[B]] =
-    xs.foldLeft(Future.successful(Queue.empty[B])) {
-      case (r, curr) =>
-        for {
-          xs <- r
-          x  <- f(curr)
-        } yield xs.enqueue(x)
-    }
+//  "Auto cancel" - {
+//    // TODO: Uncomment after Node v1.1.6
```
Comment (msmolyakov) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/CancelOrderTestSuite.scala]: move the suite to separated PR

File: dex-it/src/test/scala/com/wavesplatform/it/sync/orders/OrderDynamicFeeTestSuite.scala [hunk 6]
```diff
@@ -645,6 +645,8 @@ class OrderFeeTestSuite extends MatcherSuiteBase {
   }
 
   "rounding fee to filled amount" - {
+    val matcherFee = 300000
```
Comment (msmolyakov) [inline dex-it/src/test/scala/com/wavesplatform/it/sync/orders/OrderDynamicFeeTestSuite.scala]: use the global fee constant (from package or base suite)

Comment (ngadiyak) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/orders/OrderDynamicFeeTestSuite.scala]: ok

File: dex-it/src/test/scala/com/wavesplatform/it/sync/orders/OrderFixedFeeTestSuite.scala [hunk 7]
```diff
@@ -0,0 +1,250 @@
...
+          node.waitOrderInBlockchain(aliceOrderIdFill)
+
+          node.assetBalance(aliceAddress, fixedAssetMatcherFeeBase58).balance shouldBe aliceAssetBalanceBefore - minMatcherFee + orderAmount
+          node.assetBalance(bobAddress, fixedAssetMatcherFeeBase58).balance shouldBe bobAssetBalanceBefore - minMatcherFee - orderAmount
+          node.balanceDetails(aliceAddress).available shouldBe aliceWavesBalanceBefore - (orderAmount * price / 100000000)
+          node.balanceDetails(bobAddress).available shouldBe bobWavesBalanceBefore + (orderAmount * price / 100000000)
+        }
+      }
+
+      "should accept orders if sender received amount > then fee amount" in {
```
[the remark this answers is not in the document]
Comment (ngadiyak) [reply dex-it/src/test/scala/com/wavesplatform/it/sync/orders/OrderFixedFeeTestSuite.scala]: ok

**Pull request merged**